mllib/src/main/scala/org/apache/spark/ml/evaluation/MulticlassClassificationEvaluator.scala

```diff
@@ -38,17 +38,39 @@ class MulticlassClassificationEvaluator @Since("1.5.0") (@Since("1.5.0") overrid
   @Since("1.5.0")
   def this() = this(Identifiable.randomUID("mcEval"))
 
+  /**
+   * param for specific label value for metric to be used in evaluation (supports
+   *  Double values for metric `"f1"`, `"precision"`, `"recall"`, `"tpr"`, `"fpr"`)
+   * @group param
+   */
+  val labelValue: Param[Double] = {
+    new DoubleParam(this, "labelValue", "specific labelValue for metric to be used in evaluation")
+  }
+
+  /** @group getParam */
+  def getLabelValue: Double = $(labelValue)
+
+  /** @group setParam */
+  def setLabelValue(labelClass: Double): this.type = set(labelValue, labelClass)
+
   /**
    * param for metric name in evaluation (supports `"f1"` (default), `"weightedPrecision"`,
-   * `"weightedRecall"`, `"accuracy"`)
+   * `"weightedRecall"`, `"accuracy"` and with labelValue `"f1"`, `"precision"`, `"recall"`,
+   * `"tpr"`, `"fpr"`)
    * @group param
    */
   @Since("1.5.0")
   val metricName: Param[String] = {
-    val allowedParams = ParamValidators.inArray(Array("f1", "weightedPrecision",
-      "weightedRecall", "accuracy"))
-    new Param(this, "metricName", "metric name in evaluation " +
-      "(f1|weightedPrecision|weightedRecall|accuracy)", allowedParams)
+
+    val allowedParams: String => Boolean = { (p: String) =>
+      if (isSet(labelValue)) {
+        ParamValidators.inArray(MulticlassClassificationEvaluator.labelOptions)(p)
+      } else ParamValidators.inArray(MulticlassClassificationEvaluator.weightedOptions)(p)
+    }
```

Comment:
If we set `metricName` first, and then set `labelValue`, then it escape the check here. Does it need also add check for `labelValue` param ?